examples/spring-web/src/main/resources/static/assets/script.js

```diff
@@ -0,0 +1,45 @@
+(async function() {
+  // Clear any state in the URL
+  history.pushState({}, "", "/");
+
+  // General API request handler
+  async function get(uri) {
+    return fetch(uri).then(res => {
+      if (res.ok) {
+        return res.headers.get("Content-Type") === "application/json" ? res.json() : res.text();
+      }
+      throw new Error("Error fetching data");
+    });
+  }
+
+  // Specific API methods
+  async function getUser() {
```

Comment:
```suggestion
  function getUser() {
```

No need since no `await`s.
Returns a promise.